test/jdk/javax/swing/JFileChooser/FileViewNPETest.java

```diff
@@ -22,81 +22,110 @@
  */
 
 import java.awt.BorderLayout;
-
+import java.awt.Component;
+import java.awt.Container;
+import java.awt.Robot;
 import java.io.File;
-import java.lang.reflect.InvocationTargetException;
+import java.util.function.Predicate;
 
-import javax.swing.JFrame;
+import javax.swing.JComboBox;
 import javax.swing.JFileChooser;
+import javax.swing.JFrame;
 import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
 import javax.swing.WindowConstants;
-
 import javax.swing.filechooser.FileView;
+import javax.swing.plaf.metal.MetalLookAndFeel;
 
 /*
  * @test
  * @bug 6616245
  * @key headful
- * @requires (os.family == "windows" | os.family == "linux")
- * @library /java/awt/regtesthelpers
- * @build PassFailJFrame
  * @summary Test to check if NPE occurs when using custom FileView.
- * @run main/manual FileViewNPETest
+ * @run main FileViewNPETest
  */
 public class FileViewNPETest {
-    static PassFailJFrame passFailJFrame;
+
+    private static JFrame frame;
+    private static JFileChooser jfc;
+    private static File path;
+
     public static void main(String[] args) throws Exception {
-        SwingUtilities.invokeAndWait(new Runnable() {
-            public void run() {
-                try {
-                    initialize();
-                } catch (InterruptedException | InvocationTargetException e) {
-                    throw new RuntimeException(e);
+        final Robot robot = new Robot();
+        try {
+            SwingUtilities.invokeAndWait(FileViewNPETest::initialize);
+            robot.waitForIdle();
+
+            SwingUtilities.invokeAndWait(() -> {
+                jfc.setCurrentDirectory(path.getParentFile());
+                assert jfc.getCurrentDirectory() == null
```

Comment:
This assertion could be skipped if the test executed standalone or via directly by the jtreg w/o makefiles.